minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java

```diff
@@ -254,6 +254,25 @@ public void stop(ServerType server) throws IOException {
     stop(server, null);
   }
 
+  public void stopCompactorGroup(String compactorResourceGroup) throws IOException {
+    synchronized (compactorProcesses) {
+      var group = compactorProcesses.get(compactorResourceGroup);
+      if (group == null) {
+        return;
+      }
+      group.forEach(process -> {
+        try {
+          cluster.stopProcessWithTimeout(process, 30, TimeUnit.SECONDS);
+        } catch (ExecutionException | TimeoutException e) {
+          log.warn("Compactor did not fully stop after 30 seconds", e);
```

Comment:
This is following the pattern of other code, so I think this is fine here.  This general pattern seems like it could cause trouble.  If a test wants to stop something and runs into this and fails later because it was not stopped could make the test harder to debug.

Reply:
> This is following the pattern of other code, so I think this is fine here. This general pattern seems like it could cause trouble. If a test wants to stop something and runs into this and fails later because it was not stopped could make the test harder to debug.

I agree that it wouldn't be immediately obvious that the shutdown failing caused an error. 

To help with that, we could either throw an exception on failed process stop or this method could return an exit code. 

I'm leaning towards the exit code route since it cuts down on the amount of try/catch statements in a given test and also allows the test to ignore the output of this method if they wanted.

Do you have an opinion?

Reply:
If there are frequent cases of things wanting to ignore a failure to stop then a returning an exit code makes sense.  If not then throwing a runtime exception would cause things to fail and not introduce new try/catch blocks in the code.  Would not want to throw a checked exception, that seems like it would be annoying for this case w/o any benefit.